pkg/controller/v1alpha1/llmisvc/scheduler.go

```diff
@@ -0,0 +1,481 @@
+				continue
+			}
+
+			if slices.Contains(d.Spec.Template.Spec.Containers[i].Args, "--configText") ||
+				slices.Contains(d.Spec.Template.Spec.Containers[i].Args, "-configText") ||
+				slices.Contains(d.Spec.Template.Spec.Containers[i].Args, "--configFile") ||
+				slices.Contains(d.Spec.Template.Spec.Containers[i].Args, "-configFile") {
+				// When the configuration is overridden, don't add/override it.
+				break
+			}
+
+			d.Spec.Template.Spec.Containers[i].Args = append(d.Spec.Template.Spec.Containers[i].Args,
+				"--configText",
+				schedulerConfigText(llmSvc),
+			)
+		}
+	}
+
+	log.FromContext(ctx).V(2).Info("Expected router scheduler deployment", "deployment", d)
+
+	return d
+}
+
+func schedulerConfigText(llmSvc *v1alpha1.LLMInferenceService) string {
+	switch {
+	case llmSvc.Spec.Prefill != nil:
+		return `
+apiVersion: inference.networking.x-k8s.io/v1alpha1
+kind: EndpointPickerConfig
+plugins:
+- type: pd-profile-handler
+  parameters:
+    threshold: 100
+- type: prefill-header-handler
+- type: prefill-filter
+- type: decode-filter
+- type: prefix-cache-scorer
+- type: load-aware-scorer
+- type: max-score-picker
+schedulingProfiles:
+- name: prefill
+  plugins:
+  - pluginRef: prefill-filter
+  - pluginRef: prefix-cache-scorer
+    weight: 2.0
+  - pluginRef: load-aware-scorer
+    weight: 1.0
+  - pluginRef: max-score-picker
+- name: decode
+  plugins:
+  - pluginRef: decode-filter
+  - pluginRef: prefix-cache-scorer
+    weight: 2.0
+  - pluginRef: load-aware-scorer
+    weight: 1.0
+  - pluginRef: max-score-picker
+`
+	default:
+		return `
+apiVersion: inference.networking.x-k8s.io/v1alpha1
+kind: EndpointPickerConfig
+plugins:
+- type: single-profile-handler
+- type: prefix-cache-scorer
+- type: load-aware-scorer
+- type: max-score-picker
+schedulingProfiles:
+- name: default
+  plugins:
+  - pluginRef: prefix-cache-scorer
+    weight: 2.0
+  - pluginRef: load-aware-scorer
+    weight: 1.0
+  - pluginRef: max-score-picker
+`
+	}
+}
+
+func (r *LLMISVCReconciler) expectedSchedulerServiceAccount(llmSvc *v1alpha1.LLMInferenceService) *corev1.ServiceAccount {
+	sa := &corev1.ServiceAccount{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      kmeta.ChildName(llmSvc.GetName(), "-epp-sa"),
+			Namespace: llmSvc.GetNamespace(),
+			OwnerReferences: []metav1.OwnerReference{
+				*metav1.NewControllerRef(llmSvc, v1alpha1.LLMInferenceServiceGVK),
+			},
+			Labels: r.schedulerLabels(llmSvc),
+		},
+	}
+
+	if llmSvc.Spec.Router != nil &&
+		llmSvc.Spec.Router.Scheduler != nil &&
+		llmSvc.Spec.Router.Scheduler.Template != nil &&
+		llmSvc.Spec.Router.Scheduler.Template.ServiceAccountName != "" {
+		sa.Name = llmSvc.Spec.Router.Scheduler.Template.ServiceAccountName
+	}
+
+	return sa
+}
+
+func (r *LLMISVCReconciler) expectedSchedulerRole(llmSvc *v1alpha1.LLMInferenceService) *rbacv1.Role {
+	role := &rbacv1.Role{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      kmeta.ChildName(llmSvc.GetName(), "-epp-role"),
```

Comment:
Why do we need to create a role per inference service? looks like it is the same set of rules.